gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java

```diff
@@ -124,6 +124,32 @@ public synchronized int read(@Nonnull byte[] buf, int offset, int length) throws
     return response;
   }
 
+  @Override
+  public void readFully(long position, byte[] buffer, int offset, int length) throws IOException {
+    streamStatistics.readFullyOperationStarted(position, length);
```

Comment:
why this is done before the validation?

Reply:
This should be the protocol - statistics are for number of time API is being called not for how many times it was successful.

Reply:
Statistics are for API calls and not for correct API calls.